durablefunctionsmonitor.dotnetisolated.core/Common/Auth.cs

```diff
@@ -334,6 +324,48 @@ private static string TryGetHubNameFromHostJson()
             }
         }
 
+        private static async Task<ClaimsPrincipal> GetClaimsPrincipal(HttpRequestData request, string userNameClaimName, string rolesClaimName)
+        {
+            // First trying the request object
+            var principal = new ClaimsPrincipal(request.Identities);
+
+            if (principal.Identity != null && principal.Identity.IsAuthenticated && principal.HasClaim(c => c.Type == userNameClaimName))
+            {
+                return principal;
+            }
+
+            // Then trying to parse the x-ms-client-principal header
+            if (request.Headers.TryGetValues("x-ms-client-principal", out var clientPrincipalHeaderValues))
+            {
+                string clientPrincipalJson = Encoding.UTF8.GetString(Convert.FromBase64String(clientPrincipalHeaderValues.Single()));
```

Comment:
Since we're doing a lot of parsing here (through Single, and FromBase64String), should we add a try-catch here to report a clean error if this fails?

Reply:
Makes sense, will do.